UnityProject/Assets/Scripts/ChemistryComponents/ReagentContainer.TransferInteraction.cs

```diff
@@ -40,6 +40,8 @@ public partial class ReagentContainer : ICheckedInteractable<HandApply>, //Trans
 		[FormerlySerializedAs("TransferMode")]
 		[SerializeField] private TransferMode transferMode = TransferMode.Normal;
 
+		[SerializeField] public bool SyringePulling;
```

Comment:
Would be better to write this as:
[field: SerializeField] public bool SyringePulling { get; private set; }
as I don't think is intended for this to be able to be edited elsewhere.

Reply:
Have fun trying to convince Bod to do this

Reply:
some of you might say is it might automatically change when you fill up the syringe in the future

Reply:
> *something I will say it might be changed in the future when you fill up your syringe it swaps to inject mode

If that's the case it would be better to either
> move the syringe pulling bool to the syringe script if their is one
> have it call a function on reagentcontainer as opposed to directly changing the bool

Reply:
I have to agree with Atner here. ReagentContainer shouldn't need to know anything about syringes. The syringe itself is the one with changing modes and can interact with ReagentContainer API to manage it.

Reply:
problem is reagent container needs to know if it's a syringe

Reply:
because one of the interactions checks if it's going to be adding or removing From it

Reply:
move that interaction to the syringe compoenent(?). The syringe should be the one that handles that and makes use of the ReagentContainer component to transfer in or out depending on the state of the syringe.

Reply:
this is like explosion code having bools to know if the grenade is a frag, EMP or bang and doing all the interaction in the explosion code.

Reply:
it's literally just a small fix to make it so you can inject slime cores easier

Reply:
it's literally tech debt. But I don't know why am I even trying, you are going to merge this anyways